public/app/features/variables/state/sharedReducer.ts

```diff
@@ -121,6 +121,15 @@ const sharedReducerSlice = createSlice({
       const { option } = action.payload.data;
       const current = { ...option, text: ensureStringValues(option?.text), value: ensureStringValues(option?.value) };
 
+      // If no value is set, default to the first avilable
+      if (!current.value) {
+        instanceState.current = { ...instanceState.options[0], selected: true };
+        instanceState.options = instanceState.options.map((option, index) =>
+          index === 0 ? instanceState.current : { ...option, selected: false }
+        );
+        return;
```

Comment:
what happens if options are an empty array? will this crash?  

Since this handler can be written mutably and immer will make it immutable. I think you first can just set all options to selected false except first and set instance.current = instance.options[0] ?

Reply:
Yeah, we need to add a check to make sure we have options.